pkg/localstore/localstore.go

```diff
@@ -670,6 +629,50 @@ func New(path string, baseKey []byte, ss storage.StateStorer, o *Options, logger
 	return db, nil
 }
 
+func (db *DB) safeInit(rootPath, sharkyBasePath string) error {
+	// create if needed
+	path := filepath.Join(rootPath, sharkyDirtyFileName)
+	if _, err := os.Stat(path); errors.Is(err, fs.ErrNotExist) {
+		// missing lock file implies a clean exit then create the file and return
+		return os.WriteFile(path, []byte{}, 0644)
+	}
+
+	db.logger.Warning("localstore sharky .DIRTY file exists: starting recovery due to previous dirty exit")
+
+	locOrErr, err := recovery(db)
+	if err != nil {
+		return err
+	}
+
+	recoverySharky, err := sharky.NewRecovery(sharkyBasePath, sharkyNoOfShards, swarm.SocMaxChunkSize)
+	if err != nil {
+		return err
+	}
+
+	for l := range locOrErr {
+		if l.err != nil {
+			return l.err
+		}
+
+		err = recoverySharky.Add(l.loc)
+		if err != nil {
+			return err
+		}
+	}
+
+	err = recoverySharky.Save()
+	if err != nil {
+		return err
+	}
+
+	err = recoverySharky.Close()
+	if err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
Any chance of getting a "done" message, possibly with the elapsed time when it finishes?   Either here or maybe on a defer and show the error or success from there?

Reply:
sure, check the PR again :)